fe/fe-core/src/main/java/com/starrocks/scheduler/history/TaskRunHistoryTable.java

```diff
@@ -226,4 +236,48 @@ public List<TaskRunStatus> lookupByTaskNames(String dbName, Set<String> taskName
         Collections.sort(result, TaskRunStatus.COMPARATOR_BY_CREATE_TIME_DESC);
         return result;
     }
+
+    public List<TaskRunStatus> lookupLastJobOfTasks(String dbName, Set<String> taskNames) {
+        final String template =
+                "WITH MaxStartRunID AS (" +
+                        "    SELECT " +
+                        "       task_name, " +
+                        "       cast(history_content_json->'startTaskRunId' as string) start_run_id" +
+                        "    FROM $tableName " +
+                        "    WHERE (task_name, create_time) IN (" +
+                        "            SELECT task_name, MAX(create_time)" +
+                        "            FROM $tableName" +
+                        "            WHERE $taskFilter" +
+                        "            GROUP BY task_name" +
+                        "    )" +
+                        " )" +
+                        " SELECT t.history_content_json" +
+                        " FROM $tableName t" +
+                        " JOIN MaxStartRunID msr" +
+                        "   ON t.task_name = msr.task_name" +
+                        "   AND cast(t.history_content_json->'startTaskRunId' as string) = msr.start_run_id" +
+                        " ORDER BY t.create_time DESC";
```

Comment:
after this, the max size that history table will output will be task's number, am I right?

Reply:
seems reseaonable.